pymtl3/datatypes/bitstructs.py

```diff
@@ -412,6 +464,18 @@ def _convert_list_to_tuple( x ):
     if not '__hash__' in cls.__dict__:
       cls.__hash__ = _mk_hash_fn( fields )
 
+  # Shunning: add __ilshift__ and _flip for update_ff
+  assert not '__ilshift__' in cls.__dict__ and not '_flip' in cls.__dict__
+
+  cls.__ilshift__, cls._flip = _mk_ff_fn( fields )
+
+  def get_field_type( cls, name ):
+    if name in cls.__bitstruct_fields__:
+      return cls.__bitstruct_fields__[ name ]
+    raise AttributeError( f"{cls} has no field '{name}'" )
+
+  cls.get_field_type = classmethod(get_field_type)
```

Comment:
For consistency you probably need to assert `get_field_type` is not in the class attributes